### PR TITLE
Hosting Configuration: Add 'Enable SSH' behind a feature flag

### LINKED ISSUE #66577: Launch WPCOM SSH: Add SSH toggle to <SFTPCard> behind feature gate
In order to make SSH accessible for internal testing, we'll need to add a SSH toggle to the `<SFTPCard>` component. For the time being, it should be restricted behind a `launch-ssh` feature gate.

Here are a couple of mockups indicating how it should work:

CleanShot 2022-08-15 at 08 24 11@2x

CleanShot 2022-08-15 at 11 07 33@2x

**Done is:**
- [ ] A new "Enable SSH" toggle is added to the `<SFTPCard>` component.
- [ ] The "Enable SSH" toggle renders in its expected "Pre-activation" state.
- [ ] The "Enable SSH" toggle renders in its expected "Post-activation" state.
- [ ] A Tracks event is logged when the user activates SSH.

### PR DESCRIPTION
**Pull request by danielbachhuber**
Fixes #66577

## Proposed Changes

For sites with a plan that includes SSH, this pull request updates the `<SFTPCard>` to mention SSH and allow it to be enabled:

image

image

Sites without SSH use the existing UI.

## Testing Instructions

1. Sandbox `public-api.wordpress.com`
2. Apply D85944-code and merge trunk to make sure the latest changes are applied.
3. Create a new Business plan site.
4. Navigate to `/hosting-config/<business-site>` and observe the updated UI.
5. Enable SSH and verify SSH access works as expected.
6. Disable SSH and verify SSH access is disabled.
7. Create a new Pro plan site.
8. Navigate to `/hosting-config/<business-site>` and observe UI limited to SFTP access.

### CODE REVIEW

File: client/state/data-layer/wpcom/sites/hosting/ssh-access.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+			onError: () => {},
+			fromApi: ( { setting } ) => setting,
+		} ),
+	],
+	[ HOSTING_SSH_ACCESS_ENABLE ]: [
+		dispatchRequest( {
+			fetch: enableSshAccess,
+			onSuccess: setSshAccess,
+			onError: () => {},
+			fromApi: ( setting ) => setting,
```
Comment (p-jackson) [inline client/state/data-layer/wpcom/sites/hosting/ssh-access.js]: These can just be left out. I've only ever seen `fromApi` used when there was some actual translation needed between the network format and the format used by Calypso.

**Review by p-jackson — COMMENTED** [review]
Preliminary review looks good 👍 
You could maybe do an optimistic update of the redux store when the user toggles, rolling back if an error occurs. That way the toggle switch would feel more responsive.

File: client/state/data-layer/wpcom/sites/hosting/ssh-access.js (hunk 1 above)
Comment (danielbachhuber) [reply client/state/data-layer/wpcom/sites/hosting/ssh-access.js]: > I've only ever seen `fromApi` used when there was some actual translation needed between the network format and the format used by Calypso.

Yep! I think I'll need to do this once my feedback is applied: D85944-code#1796466

**Comment by danielbachhuber** [comment]
> Preliminary review looks good 👍

@p-jackson Thanks!

> You could maybe do an optimistic update of the redux store when the user toggles, rolling back if an error occurs. That way the toggle switch would feel more responsive.

I went with temporarily `disabled` instead, because it impacts a secondary system: a9b5303ecf

I'd love a loading state ideally, but `<ToggleControl>` doesn't seem to have one.

**Comment by danielbachhuber** [comment]
Tracks events with 1018-gh-Automattic/tracks-events-registration and 1019-gh-Automattic/tracks-events-registration

File: client/my-sites/hosting/sftp-card/index.js [hunk 2]
```diff
@@ -189,6 +265,25 @@ export const SftpCard = ( {
...
+							{ translate(
+								'SSH stands for Secure Shell. It’s a way to perform advanced operations on your site using the command line. ' +
+									'For more information see {{supportLink}}SFTP on WordPress.com{{/supportLink}}.',
+								{
+									components: {
+										supportLink: (
+											<ExternalLink
+												icon
+												target="_blank"
+												href={ localizeUrl( 'https://wordpress.com/support/sftp/' ) }
```
Comment (sejas) [inline client/my-sites/hosting/sftp-card/index.js:279]: Should we update the support link? or at least create an issue to do it in the near future?

Comment (sejas) [inline client/my-sites/hosting/sftp-card/index.js:279]: Should we change the support URL?, or at least create an issue to change it later?

**Review by sejas — COMMENTED** [review]
The code looks good, but I couldn't complete the test plan.

I'm getting a 403 from the ssh-users API:

Screenshot 2022-08-16 at 16 22 35

Not sure what I'm doing wrong.

The credentials never appear in the UI.

File: client/my-sites/hosting/sftp-card/index.js (hunk 2 above)
Comment (danielbachhuber) [reply client/my-sites/hosting/sftp-card/index.js:279]: > Should we change the support URL?, or at least create an issue to change it later?

Yep, good catch. We'll update it when the new document is ready #66604

Comment (sejas) [reply client/my-sites/hosting/sftp-card/index.js:279]: This error appears only on calypso.live , on localhost works good !
Good job @danielbachhuber

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7477076

Thank you @danielbachhuber for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7477076

Thank you @danielbachhuber for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7477076).

**Pull request merged**